lib/ansible/plugins/callback/__init__.py

```diff
@@ -128,18 +232,46 @@ def _dump_results(self, result, indent=None, sort_keys=True, keep_invocation=Fal
         if 'exception' in abridged_result:
             del abridged_result['exception']
 
-        try:
-            jsonified_results = json.dumps(abridged_result, cls=AnsibleJSONEncoder, indent=indent, ensure_ascii=False, sort_keys=sort_keys)
-        except TypeError:
-            # Python3 bug: throws an exception when keys are non-homogenous types:
-            # https://bugs.python.org/issue25457
-            # sort into an OrderedDict and then json.dumps() that instead
-            if not OrderedDict:
-                raise
-            jsonified_results = json.dumps(OrderedDict(sorted(abridged_result.items(), key=to_text)),
-                                           cls=AnsibleJSONEncoder, indent=indent,
-                                           ensure_ascii=False, sort_keys=False)
-        return jsonified_results
+        if not serialize:
+            # Just return ``abridged_result`` without going through serialization
+            # to permit callbacks to take advantage of ``_dump_results``
+            # that want to further modify the result, or use custom serialization
+            return abridged_result
+
+        if result_format == 'json':
+            try:
+                return json.dumps(abridged_result, cls=AnsibleJSONEncoder, indent=indent, ensure_ascii=False, sort_keys=sort_keys)
+            except TypeError:
+                # Python3 bug: throws an exception when keys are non-homogenous types:
+                # https://bugs.python.org/issue25457
+                # sort into an OrderedDict and then json.dumps() that instead
+                if not OrderedDict:
+                    raise
+                return json.dumps(OrderedDict(sorted(abridged_result.items(), key=to_text)),
+                                  cls=AnsibleJSONEncoder, indent=indent,
+                                  ensure_ascii=False, sort_keys=False)
+        elif result_format == 'yaml':
+            lossy = pretty_results is not False
```

Comment:
`is True`?

Reply:
`None` is being used as a special tertiary value.  `None` or `True` effectively are truthy values here.  This is largely due to the fact that the json format by default is not "pretty".

Reply:
Probably a comment in those two places then- if I was working in the neighborhood, I might've changed that since it's not obvious... Maybe `is True or is None` would be more self-explanatory?